src/crimson/osd/osd_operations/background_recovery.cc

```diff
@@ -164,11 +172,17 @@ PglogBasedRecovery::do_recovery()
   if (pg->has_reset_since(epoch_started)) {
     return seastar::make_ready_future<bool>(false);
   }
-  return with_blocking_event<RecoveryBackend::RecoveryBlockingEvent,
-			     interruptor>([this] (auto&& trigger) {
-    return pg->get_recovery_handler()->start_recovery_ops(
-      trigger,
-      crimson::common::local_conf()->osd_recovery_max_single_start);
+  auto fut = interruptor::now();
+  if (pg->have_unfound()) {
+    fut = pg->find_unfound(epoch_started);
+  }
+  return fut.then_interruptible([this] {
+    return with_blocking_event<RecoveryBackend::RecoveryBlockingEvent,
+			       interruptor>([this] (auto&& trigger) {
+      return pg->get_recovery_handler()->start_recovery_ops(
```

Comment:
IIUC, `start_recovery_ops` returned future is carrying boolean solely to determine whether we should call `find_unfound`. 
```
return pg->get_recovery_handler()->start_recovery_ops(
  ...
).then_interruptible([this] (bool do_unfound) {
  if (do_unfound) {
    return pg->find_unfound(epoch_started);
  }
});
```
Is there any other reason to return a boolean?

Reply:
@xxhdx1985126, we should either make use of start_recovery_ops returned boolean as my previous comment suggests or omit the returned boolean into am empty future.